gossip/c_block_callbacks_test.go

```diff
@@ -864,6 +869,268 @@ func TestIsPermissible_DetectsNonPermissibleTransactions(t *testing.T) {
 	}
 }
 
+func TestRlpEncodedMaxHeaderSizeInBytes_IsAnUpperBound(t *testing.T) {
+	setToMax := func(size int) []byte {
+		b := bytes.Repeat([]byte{0xFF}, size)
+		return b
+	}
+
+	maxHash := common.Hash(setToMax(32))
+	maxAddress := common.Address(setToMax(20))
+	maxBloom := types.Bloom(setToMax(2048))
+	maxBlock := types.BlockNonce(setToMax(8))
+	maxUint64 := uint64(math.MaxUint64)
+
+	// Geth sanity check for extra defines an upper bound of 100 * 1024 bytes.
+	// Sonic uses the extra field to store time and duration.
+	extra := inter.EncodeExtraData(
+		time.Unix(math.MaxInt64, math.MaxInt64),
+		time.Duration(math.MaxInt64)*time.Nanosecond,
+	)
+
+	header := &types.Header{
+		ParentHash:       maxHash,
+		UncleHash:        maxHash,
+		Coinbase:         maxAddress,
+		Root:             maxHash,
+		TxHash:           maxHash,
+		ReceiptHash:      maxHash,
+		Bloom:            maxBloom,
+		Difficulty:       big.NewInt(math.MaxInt64),
```

Comment:
big ints could use `SetBytes` to set `setToMax(256)` since that is the byte cap rlp enforces, no?